enclave_build/src/pull.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fs::File;
+use std::path::Path;
+
+use oci_distribution::{
+    client::{Client, ClientConfig, ClientProtocol, ImageData},
+    secrets::RegistryAuth,
+};
+
+use crate::{
+    image::{self, deserialize_from_reader},
+    EnclaveBuildError, Result,
+};
+
+/// Number of accepted image layer media types.
+pub const ACCEPTED_MEDIA_TYPES_COUNT: u32 = 3;
+/// Accepted image layer media types.
+pub const ACCEPTED_MEDIA_TYPES: [&str; ACCEPTED_MEDIA_TYPES_COUNT as usize] = [
+    oci_distribution::manifest::WASM_LAYER_MEDIA_TYPE,
+    oci_distribution::manifest::IMAGE_DOCKER_LAYER_GZIP_MEDIA_TYPE,
+    oci_distribution::manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE,
+];
+
+/// Builds a client which uses the protocol given as parameter.
+///
+/// Client required for the https://github.com/krustlet/oci-distribution library API.
+///
+/// By default, the client pulls the image matching the current running architecture.
+pub fn build_client(protocol: ClientProtocol) -> Client {
+    let client_config = ClientConfig {
+        protocol,
+        ..Default::default()
+    };
+    Client::new(client_config)
+}
+
+/// Checks the DOCKER_CONFIG env variable for the path to the docker config file
+///
+/// If DOCKER_CONFIG is not set, try to get the config file from the default path
+/// {XDG_RUNTIME_DIR}/containers/auth.json or {HOME}/.docker/config.json in a Podman-like behavior
+fn get_docker_config_file() -> Result<File> {
```

Comment:
Added Podman-like behaviour for searching the credentials. After checking the `DOCKER_COFIG` env variable, the search falls back in order to `{XDG_RUNTIME_DIR}/containers/auth.json` or `{HOME}/.docker/config.json`.